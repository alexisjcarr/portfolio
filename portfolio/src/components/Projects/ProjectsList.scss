.container {
  .projects-card {
    background-color: black;
    width: 90%;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    color: yellowgreen;
    line-height: 1.6;

    h1 {
      @media (max-width: 800px) {
        font-size: 20px;
      }
    }

    .project {
      h3 {
        @media (max-width: 800px) {
          font-size: 13px;
        }
      }

      .iconP {
        font-size: 30px;

        @media (max-width: 800px) {
          font-size: 15px;
        }
      }

      p {
        margin: 10px;

        @media (max-width: 800px) {
          font-size: 10px;
        }
      }

      .project-img {
        width: 50%;
        border: 1px solid yellowgreen;

        @media (max-width: 800px) {
          width: 70%;
        }
      }

      .project-img-servable {
        width: 20%;
        border: 1px solid yellowgreen;

        @media (max-width: 800px) {
          width: 50%;
        }
      }

      .desc-stack {
        display: flex;
        justify-content: center;
        flex-wrap: wrap;

        .description {
          text-align: center;
          width: 48%;

          @media (max-width: 800px) {
            width: 100%;
          }
        }

        .stack {
          text-align: center;
          width: 48%;

          @media (max-width: 800px) {
            width: 100%;
          }
        }
      }
    }
  }
}
