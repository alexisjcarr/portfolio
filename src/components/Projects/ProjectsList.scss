.container {
    .projects-card {
        background-color: black;
        width: 90%;
        box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
        color: yellowgreen;

        h1 {
            @media (max-width: 500px) {
                font-size: 20px;
            }
        }

        .project {
            h3 {
                @media (max-width: 500px) {
                    font-size: 13px;
                }
            }

            p {
                margin: 10px;

                @media (max-width: 500px) {
                    font-size: 10px;
                }
            }

            .project-img {
                width: 50%;
                border: 1px solid yellowgreen;

                @media (max-width: 500px) {
                    width: 70%;
                }
            }

            .project-img-servable {
                width: 20%;
                border: 1px solid yellowgreen;

                @media (max-width: 500px) {
                    width: 50%;
                }
            }
        }

    }
}