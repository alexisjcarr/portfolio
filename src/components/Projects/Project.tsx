import * as React from "react";
import PropTypes from "prop-types";

import "./ProjectsList.scss";

export interface ProjectProps {
  projectName: string;
  deploymentLink?: string;
  githubRepo?: string;
  role: string;
  gif?: string;
  description: string;
  techStack: string;
  imgClass: string;
}

const Project: React.SFC<ProjectProps> = ({
  projectName,
  deploymentLink,
  githubRepo,
  role,
  gif,
  description,
  techStack,
  imgClass
}) => {
  return (
    <>
      <br />
      <div className="project">
        <h3>
          {projectName}{" "}
          <a href={deploymentLink} target="_blank" rel="noopener noreferrer">
            <i className="iconP fas fa-link"></i>
          </a>{" "}
          <a href={githubRepo} target="_blank" rel="noopener noreferrer">
            <i className="iconP fab fa-github"></i>
          </a>
        </h3>
        <p>
          <span style={{ textDecoration: "underline" }}>Role:</span> {role}
        </p>
        <a href={deploymentLink} target="_blank" rel="noopener noreferrer">
          <img src={gif} alt="app" className={imgClass} />
        </a>
        <div className="desc-stack">
          <p className="description">
            <p style={{ textDecoration: "underline" }}>Description: </p>
            {description}
          </p>
          <p className="stack">
            <p style={{ textDecoration: "underline" }}>Tech Stack: </p>
            {techStack}
          </p>
        </div>
      </div>
      <br />
    </>
  );
};

Project.propTypes = {
  deploymentLink: PropTypes.string,
  githubRepo: PropTypes.string,
  role: PropTypes.string.isRequired,
  gif: PropTypes.string,
  description: PropTypes.string.isRequired,
  techStack: PropTypes.string.isRequired,
  imgClass: PropTypes.string.isRequired
};

export default Project;
