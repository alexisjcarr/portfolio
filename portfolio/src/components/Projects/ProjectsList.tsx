import * as React from 'react'

import Project from './Project'

import sleeptrackr from '../../assets/images/sleeptrackr.gif'
import fitness from '../../assets/images/fitness.gif'
import lol from '../../assets/images/portf.gif'
import sauti from '../../assets/images/ogImage.jpeg'
import insuline from '../../assets/images/insuline.gif'
import zipy from '../../assets/images/zipy.jpg'
import poster from '../../assets/images/poster.png'
import cartoon from "../../assets/images/cartoon.jpeg"
import wip from "../../assets/images/coming_soon.jpeg"
import abb from '../../assets/images/abb.gif'

import './ProjectsList.scss'

const ProjectsList: React.SFC = () => {
  const sautiDesc =
    "Sauti Africa is a non-profit organization that serves to empower cross border traders by providing real-time market prices, exchange rates, and trade procedures. This web app provides a public-facing data platform that allows for users to access Sauti Africa's market price data. "

  return (
    <div className="container" id="projects">
      <div className="projects-card">
        <h1>projects</h1>
        <hr />
        <Project
          projectName="Solutions Architecture Samples"
          role="Solutions/Systems Architect"
          description="A collection of systems/solutions architecture write-ups"
          deploymentLink="/designs"
          gif={cartoon}
          techStack="AWS, draw.io"
          imgClass="project-img"
        />
        <hr />
        <Project
          projectName="QUIC-gRPC"
          role="Backend Engineer"
          description="Still under development. gRPC is a modern open source high performance Remote Procedure Call (RPC) framework that can run in any environment. The current implementations of gRPC use HTTP/2 as the transport protocol. This project aims to replace HTTP/2 with QUIC in the Go implementation of gRPC."
          gif={wip}
          techStack="Go"
          imgClass="project-img"
        />
        <hr />
        <Project 
          projectName='Angry Beanie Babies'
          githubRepo="https://github.com/alexisjcarr/angrybeaniebabies"
          role='Game Developer'
          description='Still under development. A simple two player game where each of the two players tries to shoot down the other one. Currently working on refactoring and making an AI player for a single player experience.'
          gif={abb}
          techStack='Python, pygame'
          imgClass='project-img'
        />
        <hr />
        <Project
          projectName="this portfolio ;)"
          deploymentLink="https://www.alexisjcarr.com/"
          githubRepo="https://github.com/alexisjcarr/portfolio"
          role="Front End Engineer"
          gif={lol}
          description="A software engineer portfolio that was coded from scratch"
          techStack="React | React Hooks | TypeScript"
          imgClass="project-img"
        />
        <hr />
        <Project
          projectName="Sauti Africa Market Price Data API"
          deploymentLink="https://price-api.sautiafrica.org/"
          githubRepo="https://github.com/orgs/Lambda-School-Labs/teams/labs-17-sauti-africa-market-price-data/repositories"
          role="Fullstack Software Engineer"
          gif={sauti}
          description={sautiDesc}
          techStack="React | React Hooks | Context API | Redux | Node | Express | Redis | Auth0 | MySQL | PostgreSQL"
          imgClass="project-img"
        />
        <hr />
        {/* <Project
          projectName="macrolimiter"
          deploymentLink="https://www.npmjs.com/package/macrolimiter"
          githubRepo="https://github.com/alexisjcarr/macrolimiter"
          role="Backend Engineer"
          gif={npm}
          description="An npm package that facilitates both the generation and validation of API keys, as well as the limiting of calls for specific API keys on key-protected routes."
          techStack="Node | Express | Redis"
          imgClass="project-img"
        />
        <hr /> */}
        <Project
          projectName="Sleep Tracker"
          deploymentLink="https://sleeptrackr.netlify.com"
          githubRepo="https://github.com/alexisjcarr/Sleep-Tracker-Frontend"
          role="Front End Engineer"
          gif={sleeptrackr}
          description="A web app that recommends how much sleep you need using user inputs for mood and time slept."
          techStack="React | React Router | Redux | Material UI | Axios"
          imgClass="project-img"
        />
        <hr />
        <Project
          projectName="Grit 'N Grind"
          // deploymentLink="https://gritngrind.fitness/"
          githubRepo="https://github.com/grit-n-grind/client"
          role="Front End Engineer"
          gif={fitness}
          description="The first web application of its kind: Grit-N-Grind is a fitness
          competition application that gyms can use to throw competitions at
          their local gyms."
          techStack="React | React Hooks | React Context | Firebase"
          imgClass="project-img"
        />
        <hr />
        {/* <Project
          projectName="Servable"
          role="Fullstack Software Engineer"
          gif={servable}
          description="Still under development. A mobile-friendly web app that matches
          users to volunteer opportunities via a Tinder-like interface."
          techStack="React | React Hooks | Typescript | Node | Express | PostgreSQL"
          imgClass="project-img-servable"
        />
        <hr /> */}
        {/* <Project
          projectName="Real-time Data Renderer"
          deploymentLink="http://realtime-viz-practice.surge.sh/"
          githubRepo="https://github.com/alexisjcarr/socket-io-playground-client"
          role="Front End Engineer | Data Viz Engineer"
          gif={demo}
          description="TBA"
          techStack="React | React Hooks | SocketIO"
          imgClass="project-img"
        />
        <hr /> */}
        <Project
          projectName="Insuline (Diabetes Manager)"
          deploymentLink="https://diabetesmanager.netlify.com/"
          role="Project Lead"
          gif={insuline}
          description="A web app that allows users to input glucose levels and predicts glucose levels over the next 24 hours."
          techStack="React | React Hooks | Context API | Redux | Node | Express | SQLite | PostgreSQL | Python | Flask"
          imgClass="project-img"
        />
        <hr />
        <Project
          projectName="Brain-Machine Interface Algorithm"
          deploymentLink="https://drive.google.com/uc?exportexport=download&id=1xTlCt97Mcc4zZnFkheu11CUqovmVDO3R"
          role="Biomedical Algorithm Engineer"
          gif={poster}
          description="An algorithm that improves the accuracy of the decoder (the software that translates EEG signals into physical prosthetic movement) in an existing brain-machine interface system."
          techStack="MATLAB"
          imgClass="project-img"
        />
        <hr />
        <Project
          projectName="ZiPy (Zero-Inflated Poisson Regression Python Package)"
          githubRepo="https://github.com/alexisjcarr/ZiPy"
          role="Data Scientist/Computational Statistician"
          gif={zipy}
          description="A Python package for maximum likelihood estimation of zero-inflated models for count data."
          techStack="Python | NumPy | SciPy | Pandas"
          imgClass="project-img"
        />
      </div>
    </div>
  )
}

export default ProjectsList
