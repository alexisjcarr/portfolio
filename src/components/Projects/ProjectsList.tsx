import * as React from "react";

import Project from "./Project";

import sleeptrackr from "../../assets/images/sleeptrackr.gif";
import demo from "../../assets/images/rtsp.gif";
import fitness from "../../assets/images/fitness.gif";
import servable from "../../assets/images/servable2.gif";
import lol from "../../assets/images/portf.gif";
import sauti from "../../assets/images/ogImage.jpeg";

import "./ProjectsList.scss";

const ProjectsList: React.SFC = () => {
  const sautiDesc =
    "Sauti Africa is a non-profit organization that serves to empower cross border traders (who are predominately women) by providing real-time market prices, exchange rates and trade procedures." +
    "\n" +
    "Sauti has done so well that other organizations, policy makers, and developers have asked for access to the data that Sauti uses for their platform. However, as it currently stands, such a public-facing data platform doesn't exist." +
    "\n" +
    "The purpose of this app is to fill this need by providing what is essentially a pivot table fed by a RESTful backend. My team will be able to provide data in a tabular/chart form for those who aren't tech-minded and a consumable API for those who are.";

  return (
    <div className="container" id="projects">
      <div className="projects-card">
        <h1>projects</h1>
        <Project
          projectName="this portfolio ;)"
          deploymentLink="https://www.alexisjcarr.com/"
          githubRepo="https://github.com/alexisjcarr/portfolio"
          role="Front End Developer"
          gif={lol}
          description="A software developer portfolio that was coded from scratch"
          techStack="React | React Hooks | TypeScript"
          imgClass="project-img"
        />
        <hr />
        <Project
          projectName="Sauti Africa Market Price Data API"
          deploymentLink="http://price-api.sautiafrica.org"
          githubRepo="https://github.com/orgs/Lambda-School-Labs/teams/labs-17-sauti-africa-market-price-data/repositories"
          role="Fullstack Software Engineer"
          gif={sauti}
          description={sautiDesc}
          techStack="React | React Hooks | Redux | Node | Express | MySQL | Firebase"
          imgClass="project-img"
        />
        <hr />
        <Project
          projectName="Sleep Tracker"
          deploymentLink="https://sleeptrackr.netlify.com"
          githubRepo="https://github.com/alexisjcarr/Sleep-Tracker-Frontend"
          role="Front End Developer"
          gif={sleeptrackr}
          description="A web app that recommends how much sleep you need using user inputs for mood and time slept."
          techStack="React | React Router | Redux | Material UI | Axios"
          imgClass="project-img"
        />
        <hr />
        <Project
          projectName="Grit 'N Grind"
          deploymentLink="https://gritngrind.fitness/"
          githubRepo="https://github.com/grit-n-grind/client"
          role="Front End Developer"
          gif={fitness}
          description="The first web application of its kind: Grit-N-Grind is a fitness
          competition application that gyms can use to throw competitions at
          their local gyms."
          techStack="React | React Hooks | React Context | Firebase"
          imgClass="project-img"
        />
        <hr />
        <Project
          projectName="Servable"
          role="Fullstack Software Engineer"
          gif={servable}
          description="Still under development. A mobile-friendly web app that matches
          users to volunteer opportunities via a Tinder-like interface."
          techStack="React | React Hooks | Typescript | Node | Express | PostgreSQL"
          imgClass="project-img-servable"
        />
        <hr />
        <Project
          projectName="Real-time Data Renderer"
          deploymentLink="http://realtime-viz-practice.surge.sh/"
          githubRepo="https://github.com/alexisjcarr/socket-io-playground-client"
          role="Front End Developer | Data Viz Engineer"
          gif={demo}
          description="TBA"
          techStack="React | React Hooks | SocketIO"
          imgClass="project-img"
        />
      </div>
    </div>
  );
};

export default ProjectsList;
