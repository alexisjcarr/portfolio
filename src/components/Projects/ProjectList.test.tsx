import React from "react";
import { render, cleanup } from "@testing-library/react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import ProjectsList from "./ProjectsList";
import Project from "./Project";

configure({ adapter: new Adapter() });

describe("<ProjectsList />", () => {
  let wrapper,
    originalWarning: {
      (message?: any, ...optionalParams: any[]): void;
      (message?: any, ...optionalParams: any[]): void;
      (message?: any, ...optionalParams: any[]): void;
      (message?: any, ...optionalParams: any[]): void;
    },
    originalError: {
      (message?: any, ...optionalParams: any[]): void;
      (message?: any, ...optionalParams: any[]): void;
      (message?: any, ...optionalParams: any[]): void;
      (message?: any, ...optionalParams: any[]): void;
    };

  beforeEach(() => {
    wrapper = shallow(<ProjectsList />);

    originalWarning = console.warn;
    originalError = console.error;
    console.error = jest.fn();
    console.warn = jest.fn(); // temporarily sets console.warn to a jest function
  });

  afterEach(() => {
    cleanup();

    console.warn = originalWarning; // turns it back into the original error after test runs
    console.error = originalError;
  });

  it("matches snapshot", () => {
    expect(render(<ProjectsList />)).toMatchSnapshot();
  });

  it("contains four(4) Projects", () => {
    expect(wrapper.find(Project)).toHaveLength(4);
  });
});
