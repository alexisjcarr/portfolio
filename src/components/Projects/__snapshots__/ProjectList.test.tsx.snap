// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectsList /> matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        id="projects"
      >
        <div
          class="projects-card"
        >
          <h1>
            projects
          </h1>
          <br />
          <div
            class="project"
          >
            <h3>
              this portfolio ;)
               
              <a
                href="https://www.alexisjcarr.com/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="iconP fas fa-link"
                />
              </a>
               
              <a
                href="https://github.com/alexisjcarr/portfolio"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="iconP fab fa-github"
                />
              </a>
            </h3>
            <p>
              <span
                style="text-decoration: underline;"
              >
                Role:
              </span>
               
              Front End Developer
            </p>
            <a
              href="https://www.alexisjcarr.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="app"
                class="project-img"
                src="portf.gif"
              />
            </a>
            <div
              class="desc-stack"
            >
              <p
                class="description"
              >
                <p
                  style="text-decoration: underline;"
                >
                  Description: 
                </p>
                A software developer portfolio that was coded from scratch
              </p>
              <p
                class="stack"
              >
                <p
                  style="text-decoration: underline;"
                >
                  Tech Stack: 
                </p>
                React | React Hooks | TypeScript
              </p>
            </div>
          </div>
          <br />
          <hr />
          <br />
          <div
            class="project"
          >
            <h3>
              Sauti Africa Market Price Data API
               
              <a
                href="http://price-api.sautiafrica.org"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="iconP fas fa-link"
                />
              </a>
               
              <a
                href="https://github.com/orgs/Lambda-School-Labs/teams/labs-17-sauti-africa-market-price-data/repositories"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="iconP fab fa-github"
                />
              </a>
            </h3>
            <p>
              <span
                style="text-decoration: underline;"
              >
                Role:
              </span>
               
              Fullstack Software Engineer
            </p>
            <a
              href="http://price-api.sautiafrica.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="app"
                class="project-img"
                src="ogImage.jpeg"
              />
            </a>
            <div
              class="desc-stack"
            >
              <p
                class="description"
              >
                <p
                  style="text-decoration: underline;"
                >
                  Description: 
                </p>
                Sauti Africa is a non-profit organization that serves to empower cross border traders (who are predominately women) by providing real-time market prices, exchange rates and trade procedures. Sauti has done so well that other organizations, policy makers, and developers have asked for access to the data that Sauti uses for their platform. However, as it currently stands, such a public-facing data platform doesn't exist. The purpose of this app is to fill this need by providing what is essentially a pivot table fed by a RESTful backend. My team will be able to provide data in a tabular/chart form for those who aren't tech-minded and a consumable API for those who are.
              </p>
              <p
                class="stack"
              >
                <p
                  style="text-decoration: underline;"
                >
                  Tech Stack: 
                </p>
                React | React Hooks | Redux | Node | Express | MySQL | Firebase
              </p>
            </div>
          </div>
          <br />
          <hr />
          <br />
          <div
            class="project"
          >
            <h3>
              Sleep Tracker
               
              <a
                href="https://sleeptrackr.netlify.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="iconP fas fa-link"
                />
              </a>
               
              <a
                href="https://github.com/alexisjcarr/Sleep-Tracker-Frontend"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="iconP fab fa-github"
                />
              </a>
            </h3>
            <p>
              <span
                style="text-decoration: underline;"
              >
                Role:
              </span>
               
              Front End Developer
            </p>
            <a
              href="https://sleeptrackr.netlify.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="app"
                class="project-img"
                src="sleeptrackr.gif"
              />
            </a>
            <div
              class="desc-stack"
            >
              <p
                class="description"
              >
                <p
                  style="text-decoration: underline;"
                >
                  Description: 
                </p>
                A web app that recommends how much sleep you need using user inputs for mood and time slept.
              </p>
              <p
                class="stack"
              >
                <p
                  style="text-decoration: underline;"
                >
                  Tech Stack: 
                </p>
                React | React Router | Redux | Material UI | Axios
              </p>
            </div>
          </div>
          <br />
          <hr />
          <br />
          <div
            class="project"
          >
            <h3>
              Grit 'N Grind
               
              <a
                href="https://gritngrind.fitness/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="iconP fas fa-link"
                />
              </a>
               
              <a
                href="https://github.com/grit-n-grind/client"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="iconP fab fa-github"
                />
              </a>
            </h3>
            <p>
              <span
                style="text-decoration: underline;"
              >
                Role:
              </span>
               
              Front End Developer
            </p>
            <a
              href="https://gritngrind.fitness/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="app"
                class="project-img"
                src="fitness.gif"
              />
            </a>
            <div
              class="desc-stack"
            >
              <p
                class="description"
              >
                <p
                  style="text-decoration: underline;"
                >
                  Description: 
                </p>
                The first web application of its kind: Grit-N-Grind is a fitness competition application that gyms can use to throw competitions at their local gyms.
              </p>
              <p
                class="stack"
              >
                <p
                  style="text-decoration: underline;"
                >
                  Tech Stack: 
                </p>
                React | React Hooks | React Context | Firebase
              </p>
            </div>
          </div>
          <br />
          <hr />
          <br />
          <div
            class="project"
          >
            <h3>
              Servable
               
              <a
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="iconP fas fa-link"
                />
              </a>
               
              <a
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="iconP fab fa-github"
                />
              </a>
            </h3>
            <p>
              <span
                style="text-decoration: underline;"
              >
                Role:
              </span>
               
              Fullstack Software Engineer
            </p>
            <a
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="app"
                class="project-img-servable"
                src="servable2.gif"
              />
            </a>
            <div
              class="desc-stack"
            >
              <p
                class="description"
              >
                <p
                  style="text-decoration: underline;"
                >
                  Description: 
                </p>
                Still under development. A mobile-friendly web app that matches users to volunteer opportunities via a Tinder-like interface.
              </p>
              <p
                class="stack"
              >
                <p
                  style="text-decoration: underline;"
                >
                  Tech Stack: 
                </p>
                React | React Hooks | Typescript | Node | Express | PostgreSQL
              </p>
            </div>
          </div>
          <br />
          <hr />
          <br />
          <div
            class="project"
          >
            <h3>
              Real-time Data Renderer
               
              <a
                href="http://realtime-viz-practice.surge.sh/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="iconP fas fa-link"
                />
              </a>
               
              <a
                href="https://github.com/alexisjcarr/socket-io-playground-client"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  class="iconP fab fa-github"
                />
              </a>
            </h3>
            <p>
              <span
                style="text-decoration: underline;"
              >
                Role:
              </span>
               
              Front End Developer | Data Viz Engineer
            </p>
            <a
              href="http://realtime-viz-practice.surge.sh/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="app"
                class="project-img"
                src="rtsp.gif"
              />
            </a>
            <div
              class="desc-stack"
            >
              <p
                class="description"
              >
                <p
                  style="text-decoration: underline;"
                >
                  Description: 
                </p>
                TBA
              </p>
              <p
                class="stack"
              >
                <p
                  style="text-decoration: underline;"
                >
                  Tech Stack: 
                </p>
                React | React Hooks | SocketIO
              </p>
            </div>
          </div>
          <br />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      id="projects"
    >
      <div
        class="projects-card"
      >
        <h1>
          projects
        </h1>
        <br />
        <div
          class="project"
        >
          <h3>
            this portfolio ;)
             
            <a
              href="https://www.alexisjcarr.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="iconP fas fa-link"
              />
            </a>
             
            <a
              href="https://github.com/alexisjcarr/portfolio"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="iconP fab fa-github"
              />
            </a>
          </h3>
          <p>
            <span
              style="text-decoration: underline;"
            >
              Role:
            </span>
             
            Front End Developer
          </p>
          <a
            href="https://www.alexisjcarr.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="app"
              class="project-img"
              src="portf.gif"
            />
          </a>
          <div
            class="desc-stack"
          >
            <p
              class="description"
            >
              <p
                style="text-decoration: underline;"
              >
                Description: 
              </p>
              A software developer portfolio that was coded from scratch
            </p>
            <p
              class="stack"
            >
              <p
                style="text-decoration: underline;"
              >
                Tech Stack: 
              </p>
              React | React Hooks | TypeScript
            </p>
          </div>
        </div>
        <br />
        <hr />
        <br />
        <div
          class="project"
        >
          <h3>
            Sauti Africa Market Price Data API
             
            <a
              href="http://price-api.sautiafrica.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="iconP fas fa-link"
              />
            </a>
             
            <a
              href="https://github.com/orgs/Lambda-School-Labs/teams/labs-17-sauti-africa-market-price-data/repositories"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="iconP fab fa-github"
              />
            </a>
          </h3>
          <p>
            <span
              style="text-decoration: underline;"
            >
              Role:
            </span>
             
            Fullstack Software Engineer
          </p>
          <a
            href="http://price-api.sautiafrica.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="app"
              class="project-img"
              src="ogImage.jpeg"
            />
          </a>
          <div
            class="desc-stack"
          >
            <p
              class="description"
            >
              <p
                style="text-decoration: underline;"
              >
                Description: 
              </p>
              Sauti Africa is a non-profit organization that serves to empower cross border traders (who are predominately women) by providing real-time market prices, exchange rates and trade procedures. Sauti has done so well that other organizations, policy makers, and developers have asked for access to the data that Sauti uses for their platform. However, as it currently stands, such a public-facing data platform doesn't exist. The purpose of this app is to fill this need by providing what is essentially a pivot table fed by a RESTful backend. My team will be able to provide data in a tabular/chart form for those who aren't tech-minded and a consumable API for those who are.
            </p>
            <p
              class="stack"
            >
              <p
                style="text-decoration: underline;"
              >
                Tech Stack: 
              </p>
              React | React Hooks | Redux | Node | Express | MySQL | Firebase
            </p>
          </div>
        </div>
        <br />
        <hr />
        <br />
        <div
          class="project"
        >
          <h3>
            Sleep Tracker
             
            <a
              href="https://sleeptrackr.netlify.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="iconP fas fa-link"
              />
            </a>
             
            <a
              href="https://github.com/alexisjcarr/Sleep-Tracker-Frontend"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="iconP fab fa-github"
              />
            </a>
          </h3>
          <p>
            <span
              style="text-decoration: underline;"
            >
              Role:
            </span>
             
            Front End Developer
          </p>
          <a
            href="https://sleeptrackr.netlify.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="app"
              class="project-img"
              src="sleeptrackr.gif"
            />
          </a>
          <div
            class="desc-stack"
          >
            <p
              class="description"
            >
              <p
                style="text-decoration: underline;"
              >
                Description: 
              </p>
              A web app that recommends how much sleep you need using user inputs for mood and time slept.
            </p>
            <p
              class="stack"
            >
              <p
                style="text-decoration: underline;"
              >
                Tech Stack: 
              </p>
              React | React Router | Redux | Material UI | Axios
            </p>
          </div>
        </div>
        <br />
        <hr />
        <br />
        <div
          class="project"
        >
          <h3>
            Grit 'N Grind
             
            <a
              href="https://gritngrind.fitness/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="iconP fas fa-link"
              />
            </a>
             
            <a
              href="https://github.com/grit-n-grind/client"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="iconP fab fa-github"
              />
            </a>
          </h3>
          <p>
            <span
              style="text-decoration: underline;"
            >
              Role:
            </span>
             
            Front End Developer
          </p>
          <a
            href="https://gritngrind.fitness/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="app"
              class="project-img"
              src="fitness.gif"
            />
          </a>
          <div
            class="desc-stack"
          >
            <p
              class="description"
            >
              <p
                style="text-decoration: underline;"
              >
                Description: 
              </p>
              The first web application of its kind: Grit-N-Grind is a fitness competition application that gyms can use to throw competitions at their local gyms.
            </p>
            <p
              class="stack"
            >
              <p
                style="text-decoration: underline;"
              >
                Tech Stack: 
              </p>
              React | React Hooks | React Context | Firebase
            </p>
          </div>
        </div>
        <br />
        <hr />
        <br />
        <div
          class="project"
        >
          <h3>
            Servable
             
            <a
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="iconP fas fa-link"
              />
            </a>
             
            <a
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="iconP fab fa-github"
              />
            </a>
          </h3>
          <p>
            <span
              style="text-decoration: underline;"
            >
              Role:
            </span>
             
            Fullstack Software Engineer
          </p>
          <a
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="app"
              class="project-img-servable"
              src="servable2.gif"
            />
          </a>
          <div
            class="desc-stack"
          >
            <p
              class="description"
            >
              <p
                style="text-decoration: underline;"
              >
                Description: 
              </p>
              Still under development. A mobile-friendly web app that matches users to volunteer opportunities via a Tinder-like interface.
            </p>
            <p
              class="stack"
            >
              <p
                style="text-decoration: underline;"
              >
                Tech Stack: 
              </p>
              React | React Hooks | Typescript | Node | Express | PostgreSQL
            </p>
          </div>
        </div>
        <br />
        <hr />
        <br />
        <div
          class="project"
        >
          <h3>
            Real-time Data Renderer
             
            <a
              href="http://realtime-viz-practice.surge.sh/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="iconP fas fa-link"
              />
            </a>
             
            <a
              href="https://github.com/alexisjcarr/socket-io-playground-client"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="iconP fab fa-github"
              />
            </a>
          </h3>
          <p>
            <span
              style="text-decoration: underline;"
            >
              Role:
            </span>
             
            Front End Developer | Data Viz Engineer
          </p>
          <a
            href="http://realtime-viz-practice.surge.sh/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="app"
              class="project-img"
              src="rtsp.gif"
            />
          </a>
          <div
            class="desc-stack"
          >
            <p
              class="description"
            >
              <p
                style="text-decoration: underline;"
              >
                Description: 
              </p>
              TBA
            </p>
            <p
              class="stack"
            >
              <p
                style="text-decoration: underline;"
              >
                Tech Stack: 
              </p>
              React | React Hooks | SocketIO
            </p>
          </div>
        </div>
        <br />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
